src/libraries/Common/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -172,6 +173,63 @@ public ZLibStreamHandle()
                 SetHandle(IntPtr.Zero);
             }
 
+            public static ZLibStreamHandle CreateForDeflate(CompressionLevel level, int windowBits,
+                int memLevel, CompressionStrategy strategy)
+            {
+                ZLibStreamHandle zLibStreamHandle = new ZLibStreamHandle();
+                ErrorCode errC;
+
+                try
+                {
+                    errC = zLibStreamHandle.DeflateInit2_(level, windowBits, memLevel, strategy);
+                }
+                catch (Exception cause) // could not load the ZLib dll
+                {
+                    zLibStreamHandle.Dispose();
+                    throw new ZLibNativeException(cause);
+                }
+
+                if (errC == ErrorCode.Ok)
```

Comment:
Nit: I'd prefer to see these checks inverted, e.g.
```C#
if (errC is not ErrorCode.Ok)
{
    ...
    throw ...;
}

return zLibStreamHandle;
```

Reply:
This has been fixed as of 50d5f8d07fadcf122b3d45400ab9665c690feacd, and I don't see any other occurrences of the same pattern. The other open comments are functionally done, and just need a review.